pkg/recommendation/device.go

```diff
@@ -0,0 +1,235 @@
+package recommendation
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"mayadata.io/cstorpoolauto/types"
+	bdutil "mayadata.io/cstorpoolauto/util/blockdevice"
+)
+
+// nodeCapacityBlockDevice contains key with node name and value with capacityBlockDevice
+type nodeCapacityBlockDevice map[string]capacityBlockDevices
+
+// capacityBlockDevices contains a key with capacity of block device and
+// value with all the block devices of that capacity.
+type capacityBlockDevices map[int64][]bdutil.MetaInfo
+
+type cStorPoolClusterRecommendationRequest types.CStorPoolClusterRecommendationRequest
```

Comment:
What is the main reason for this type alias?

Reply:
should we have a sub package under `types`?
e.g. types/cstorpoolcluster/types.go

This will ensure the client code looks like below:
```go
import (
 typecspc "mayadata-io/cstorpoolauto/types/cstorpoolcluster"
)

// a sample usage that is not too long
// req := &typecspc.RecommendationRequest{}

```

Reply:
cc @shovanmaity